apps/server/src/infra/calendar/service/calendar.service.ts

```diff
@@ -1,26 +1,85 @@
 import { Configuration } from '@hpi-schul-cloud/commons/lib';
 import { HttpService } from '@nestjs/axios';
 import { Injectable, InternalServerErrorException } from '@nestjs/common';
-import { EntityId } from '@shared/domain/types';
 import { ErrorUtils } from '@src/core/error/utils';
 import { AxiosRequestConfig, AxiosResponse } from 'axios';
 import { firstValueFrom, Observable } from 'rxjs';
 import { URL, URLSearchParams } from 'url';
+import { Logger } from '@src/core/logger';
+import { EventBus, IEventHandler } from '@nestjs/cqrs';
+import {
+	DataDeletedEvent,
+	DataDeletionDomainOperationLoggable,
+	DeletionService,
+	DomainDeletionReport,
+	DomainDeletionReportBuilder,
+	DomainName,
+	DomainOperationReportBuilder,
+	OperationType,
+	StatusModel,
+	UserDeletedEvent,
+} from '@src/modules/deletion';
+import { EntityId } from '@shared/domain/types';
 import { CalendarEventDto } from '../dto/calendar-event.dto';
 import { CalendarEvent } from '../interface/calendar-event.interface';
 import { CalendarMapper } from '../mapper/calendar.mapper';
 
 @Injectable()
-export class CalendarService {
+export class CalendarService implements DeletionService, IEventHandler<UserDeletedEvent> {
 	private readonly baseURL: string;
 
 	private readonly timeoutMs: number;
 
-	constructor(private readonly httpService: HttpService, private readonly calendarMapper: CalendarMapper) {
+	constructor(
+		private readonly httpService: HttpService,
+		private readonly calendarMapper: CalendarMapper,
+		private readonly logger: Logger,
+		private readonly eventBus: EventBus
+	) {
+		this.logger.setContext(CalendarService.name);
 		this.baseURL = Configuration.get('CALENDAR_URI') as string;
 		this.timeoutMs = Configuration.get('REQUEST_OPTION__TIMEOUT_MS') as number;
 	}
 
+	public async handle({ deletionRequestId, targetRefId }: UserDeletedEvent): Promise<void> {
+		const dataDeleted = await this.deleteUserData(targetRefId);
+		await this.eventBus.publish(new DataDeletedEvent(deletionRequestId, dataDeleted));
+	}
+
+	public async deleteUserData(userId: EntityId): Promise<DomainDeletionReport> {
+		this.logger.info(
+			new DataDeletionDomainOperationLoggable(
+				'Deleting data from Calendar Service',
+				DomainName.CALENDAR,
+				userId,
+				StatusModel.PENDING
+			)
+		);
+
+		if (!userId) {
+			throw new InternalServerErrorException('User id is missing');
+		}
+
+		await this.deleteEventsByScopeId(userId);
+
+		const result = DomainDeletionReportBuilder.build(DomainName.CALENDAR, [
+			DomainOperationReportBuilder.build(OperationType.DELETE, Number.NaN, [userId]),
```

Comment:
we should log the number of deleted events, proposes to enter 0 now, and in another ticket we will add a reliable method to verify the number of deleted events